phoenix-core/src/main/java/org/apache/phoenix/iterate/ExplainTable.java

```diff
@@ -302,7 +306,8 @@ private void appendPKColumnValue(StringBuilder buf, byte[] range, Boolean isNull
     private Long getViewIndexValue(PDataType type, byte[] range) {
         boolean useLongViewIndex = MetaDataUtil.getViewIndexIdDataType().equals(type);
         Object s = type.toObject(range);
-        return (useLongViewIndex ? (Long) s : (Short) s) + Short.MAX_VALUE + 2;
+        //return (useLongViewIndex ? (Long) s : (Short) s) + Short.MAX_VALUE + 2;
+        return (useLongViewIndex ? (Long) s : 0) + Short.MAX_VALUE + 2;
```

Comment:
I suspect this will cause some very tricky backwards compatibility issue.

Reply:
Good catch! I forgot to undue the change there. I will fix it.